applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py

```diff
@@ -260,37 +260,39 @@ def GetMinimumBufferSize(self):
         return self.min_buffer_size
 
     def AddDofs(self):
-        settings = self.main_model_part.ProcessInfo[KratosMultiphysics.CONVECTION_DIFFUSION_SETTINGS]
+        # Set DOFs and reaction variables list from Kratos parameters settings
+        dofs_with_reactions_list = []
+        conv_diff_vars = self.settings["convection_diffusion_variables"]
+        dof_var_name = conv_diff_vars["unknown_variable"].GetString()
+        reaction_var_name = conv_diff_vars["reaction_variable"].GetString()
+        dofs_with_reactions_list.append([dof_var_name,reaction_var_name])
+        if self.settings["gradient_dofs"].GetBool():
+            grad_dof_var_name = conv_diff_vars["gradient_variable"].GetString()
+            grad_react_var_name = conv_diff_vars["reaction_gradient_variable"].GetString()
+            comp_list = ["_X","_Y"] if self.settings["domain_size"].GetInt() == 2 else ["_X","_Y","_Z"]
+            for comp in comp_list:
+                dofs_with_reactions_list.append([grad_dof_var_name+comp,grad_react_var_name+comp])
 
-        # Add standard scalar DOF
-        if settings.IsDefinedReactionVariable():
-            KratosMultiphysics.VariableUtils().AddDof(settings.GetUnknownVariable(), settings.GetReactionVariable(),self.main_model_part)
-        else:
-            KratosMultiphysics.VariableUtils().AddDof(settings.GetUnknownVariable(), self.main_model_part)
+        # Add the DOFs and reaction list to each node
+        KratosMultiphysics.VariableUtils.AddDofsList(dofs_with_reactions_list, self.main_model_part)
+
+        KratosMultiphysics.Logger.PrintInfo("::[ConvectionDiffusionSolver]:: ", "DOF's ADDED")
```

Comment:
I would simplify this by using `GetDofsList` method, and adding reaction dofs afterwards.